src/datepicker.jsx

```diff
@@ -188,7 +188,7 @@ var DatePicker = React.createClass({
       return
     }
 
-    if (!isSameDay(this.props.selected, changedDate)) {
+    if (!isSameDay(this.props.selected, changedDate) || this.props.disableDateAutoCorrection) {
```

Comment:
If they enter the field and then leave without changing the value, it would not fire.  The onChange handling from the `DateInput` component (`date_input.jsx`) calls `this.props.onChangeDate` when processing a value change which is what triggers the function in this file to run.  It will either pass a valid date object or null if it's invalid.

That said, the changed file will keep calling the `onChange` function after each character change.  So if a user tabs to the field and types: 0..2../..2..2../..2..0..1..7, this function will call `onChange` 10 times.  The first 9 times will be with a null date value and the 10th time will have the valid date value.  If a user selects a date from the calendar (mouse or keyboard), it will just call `onChange` once.